File: src/services/invoice-management/Myb.Invoice.Models/InvoiceModel.cs
using Myb.Common.Models;
using System;
using System.Collections.Generic;
using System.Linq;


namespace Myb.Invoice.Models
{
    public class InvoiceModel : BaseEntity
    {
        public string? InvoiceNum { get; set; }
        public string? UserId { get; set; }
        public int? CompanyId { get; set; }
        public DateTime? InvoiceDate { get; set; }
        public DateTime? DueDate { get; set;}
        public double? TotalAmount { get; set; }
        public double? SubTotal { get; set;}
        public string? Status { get; set; }
        public int? ClientID { get; set; }
        public virtual Client? Client { get; set; }
        public virtual ICollection<InvoiceDetails>? InvoiceDetails { get; set; }
        public string? Image {  get; set; }
    }
}

File: src/front/myb.front/libs/invoice-module/src/lib/models/invoice.model.ts
import { BaseModel } from 'libs/shared/infra/models/base.model';
import { InvoiceDetails } from './invoiceDetails.model';

export class Invoice extends BaseModel {
  invoiceNum?: string;
  userId?: string;
  companyId?: number;
  invoiceDate?: Date | null;
  dueDate?: Date | null;
  totalAmount?: number;
  subTotal?: number;
  status?: string;
  clientID?: number;
  invoiceDetails?: InvoiceDetails[];
}
